src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -44,18 +42,17 @@ internal class DefaultVisualStudioRazorParser : VisualStudioRazorParser, IDispos
         private readonly ErrorReporter _errorReporter;
         private readonly List<CodeDocumentRequest> _codeDocumentRequests;
         private readonly TaskScheduler _uiThreadScheduler;
-        private RazorProjectEngine _projectEngine;
-        private RazorCodeDocument _codeDocument;
-        private ITextSnapshot _snapshot;
+        private RazorProjectEngine? _projectEngine;
+        private RazorCodeDocument? _codeDocument;
+        private ITextSnapshot? _snapshot;
         private bool _disposed;
-        private ITextSnapshot _latestParsedSnapshot;
+        private ITextSnapshot? _latestParsedSnapshot;
 
         // For testing only
         [Obsolete("Testing Only")]
-        internal DefaultVisualStudioRazorParser(RazorCodeDocument codeDocument)
-        {
-            _codeDocument = codeDocument;
-        }
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+        internal DefaultVisualStudioRazorParser(RazorCodeDocument codeDocument) => _codeDocument = codeDocument;
```

Comment:
Very very very much prefer the full bodied version.

I don't really understand why expression bodied constructors are even in the language 😛

Reply:
Agreed. I have an aversion to expression-bodied members being using with assignment expressions.